packages/components/package.json

```diff
@@ -12,9 +12,7 @@
   },
   "dependencies": {
     "@emotion/react": "^11.1.1",
-    "@emotion/styled": "^11.0.0",
```

Comment:
_index.d.ts_ imports StyledComponent and we leak it in our public API.

```ts
import { StyledComponent } from '@emotion/styled'
```
```ts
export const Box: StyledComponent<BoxOwnProps, BoxProps>
```

We'll need an update to `index.d.ts` here.

Reply:
I can update it for the sake of it, however it is imported by `@emotion/react` as a devDependency anyway:

https://github.com/emotion-js/emotion/blob/a89d4257b0246a1640a99f77838e5edad4ec4386/packages/react/package.json#L54

Reply:
>  as a devDependency anyway

It's not installed in user's node_modules then.

Reply:
Not sure what to do about it - all types are dev dependencies and they are not part of the production bundles.

1. We can add the type and pretty much duplicate `StyledComponent`, with all the risks duplicating code entails. 
2. We can add  `@emotion/styled` as a devDependency (different from using it at run-time as previously). 
3. We can leave as is.

I took #3 but  open to different opinions.

Reply:
> We can add @emotion/styled as a devDependency (different from using it at run-time as previously).
> Not sure what to do about it - all types are dev dependencies and they are not part of the production bundles.

`"dependencies"` and `"devDependencies"` have no inherent meaning in the context of bundling. If I remembere correctly, **brunch** bundler used to bundle all `"dependencies"`, but **webpack** doesn't care about them at all (It traverses your module tree.)

They have meaning for package managers however. If a package A has package B in `"dependencies"`, both will appear in user's node_modules when they do `npm install A`. This is not the case for `"devDependencies"`. In case of `"peerDependencies"`, the user will get a warning in the terminal prompting them to install the peerDependency.

If we use types from a devDependency, users won't know to install the package, but the types will collapse to `any` or fail their build when their tsconfig option `skipLibCheck` is set to false.

You can read more about it in npm docs:
- [dependencies](https://docs.npmjs.com/cli/v6/configuring-npm/package-json#dependencies)
- [devDependencies](https://docs.npmjs.com/cli/v6/configuring-npm/package-json#devdependencies)
- [peerDependencies](https://docs.npmjs.com/cli/v6/configuring-npm/package-json#peerdependencies)

There are also `bundledDependencies` and `optionalDependencies`, but I haven't seen anybody use them, so I'm not sure what's the support.

Reply:
Thanks for the review, I think it should be a devDependency (as it is already through `@emotion/react`, yes its just the typescript type and not sure its worth duplicating the same code. Once we remove the `withComponent` obsolete, we can “rewrite” our own and drop it completely

Reply:
What part of the type do we need from Emotion?

Reply:
Its just ‘StyledComponent’ 
https://github.com/system-ui/theme-ui/blob/90b7c10b2c2533c0126c9e12787997c67778eb39/packages/components/index.d.ts#L2

Reply:
> Thanks for the review, I think it should be a devDependency.

So if somebody is depending just on `theme-ui`, and doesn't have `@emotion/styled` in their own dependencies or devDependencies, this change will remove `@emotion/styled` from their node_modules.

If they use TypeScript with `skipLibCheck: false`, we'll crash their build.
If they use JavaScript or skip lib check, we'll de facto disable editor support on Box and Flex, what is still an regression.

If we remove @emotion/styled from runtime, we need to stop depending on their types, especially since the types for Box and Flex is this PR are now incorrect.

Reply:
@hasparus Will your PR to move `@emotion/styled` to user/peer deps fix this?

Reply:
@lachlanjc I started working on removing @emotion/styled on `remove-emotion-styled-2` branch — take a look at https://github.com/system-ui/theme-ui/compare/develop...remove-emotion-styled-2.

I think we'll release peer-deps branch and @emotion/styled removal in the same version — both might be breaking changes.